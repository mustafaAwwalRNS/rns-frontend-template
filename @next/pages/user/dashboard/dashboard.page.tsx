import React from "react";
import { Surface } from "@components/atoms";
import { UserPanel } from "@components/templates";
import { Notifications } from "@components/organisms";

/**
 * Dashboard
 *
 * Features:
 * - Renders user dashboard.
 */
export const Dashboard = (): JSX.Element => {
  return (
    <Surface>
      <Notifications />
    </Surface>
  );
};

Dashboard.getLayout = (page: React.ReactNode) => <UserPanel>{page}</UserPanel>;
