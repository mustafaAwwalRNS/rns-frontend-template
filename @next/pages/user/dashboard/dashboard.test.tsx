import React from "react";
import { render } from "test-utils";
import { Dashboard } from "./dashboard.page";

const setup = () => {
  const {
    container: { firstChild: dashboard },
  } = render(<Dashboard />);
  return { dashboard };
};

describe("<Dashboard />", () => {
  it("should match snapshot", () => {
    const { dashboard } = setup();
    expect(dashboard).toMatchSnapshot();
  });
});
