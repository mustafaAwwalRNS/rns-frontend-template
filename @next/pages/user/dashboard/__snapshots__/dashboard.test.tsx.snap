// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match snapshot 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  color: black;
}

.c3 {
  border-top: 1px solid #ccc;
}

.c0 {
  background-color: white;
  border-radius: 0.5rem;
  padding: 0.5rem;
  box-shadow: 0 2px 6px rgba(0,0,0,0.12),0 2px 4px rgba(0,0,0,0.24);
}

.c1 {
  background-color: white;
  border-radius: 0.5rem;
  padding: 1rem;
  box-shadow: 0 2px 6px rgba(0,0,0,0.12),0 2px 4px rgba(0,0,0,0.24);
}

<div
  class="c0"
>
  <div
    class="c1"
    spacing="2"
  >
    <h2
      class="c2"
    >
      Notifications
    </h2>
    <hr
      class="c3"
    />
  </div>
</div>
`;
